sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared_async.py

```diff
@@ -0,0 +1,64 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import asyncio
+from collections.abc import AsyncIterator
+from typing import TYPE_CHECKING, Any
+from ..pipeline import PipelineContext, PipelineRequest, PipelineResponse
+from ..pipeline._tools_async import await_result as _await_result
+
+if TYPE_CHECKING:
+    from typing import List
+    from ..pipeline._tools_async import HTTPResponseType
+    from ..pipeline.policies import SansIOHTTPPolicy
+
+class _PartGenerator(AsyncIterator):
+    """Until parts is a real async iterator, wrap the sync call.
+
+    :param parts: An iterable of parts
```

Comment:
Do we need to update this?

Reply:
do an async generator instead of having a class compatible with 3.5? I had the same thoughts, I can open an issue to do it in another PR if we think it's necessary